gymnasium/envs/mujoco/mujoco_rendering.py

```diff
@@ -356,11 +356,20 @@ def make_context_current(self):
         glfw.make_context_current(self.window)
 
     def free(self):
-        if self.window:
-            if glfw.get_current_context() == self.window:
-                glfw.make_context_current(None)
-            glfw.destroy_window(self.window)
-            self.window = None
+        """
+        Safely frees the OpenGL context and destroys the GLFW window, 
+        handling potential issues during interpreter shutdown or resource cleanup.
+        """
+        try:
+            if self.window:
+                if glfw.get_current_context() == self.window:
+                    glfw.make_context_current(None)
+                glfw.destroy_window(self.window)
+                self.window = None
+        except AttributeError:
```

Comment:
Can a more explicitly error type other than `AttributeError` be used?

If not, it is fine

Reply:
Let me look into it. Maybe there is another error which can be caught earlier in the stack trace.

Reply:
Nope. This is the exact error it throws. No other error is thrown or handled before it.

I did try to use the sys lib to handle it more gracefully, but it gets unloaded before glfw thus resulting in an import error which is more confusing.